src/main/java/com/example/report/service/ReportService.java

```diff
@@ -119,11 +143,95 @@ public List<ReportEntryDTO> generateReportEntries(LocalDate from, LocalDate to)
     }
+
+    private void sheetStyle(Sheet sheet){
+        sheet.autoSizeColumn(0);
+        sheet.autoSizeColumn(1);
+        sheet.autoSizeColumn(2);
+        sheet.createFreezePane(0,1);
+        sheet.setDisplayGridlines(false);
+        sheet.setZoom(130);
+    }
+
+    private void headerCellStyle(Workbook workbook, Sheet sheet, CellStyle headerStyle){
+        XSSFFont font = ((XSSFWorkbook) workbook).createFont();
+        font.setFontName("Arial");
+        font.setFontHeightInPoints((short) 11);
+        font.setBold(true);
+        font.setColor(HSSFColor.HSSFColorPredefined.WHITE.getIndex());
+        headerStyle.setFillForegroundColor(IndexedColors.TEAL.getIndex());
+        headerStyle.setFillPattern(FillPatternType.SOLID_FOREGROUND);
+        headerStyle.setFont(font);
+        headerStyle.setAlignment(HorizontalAlignment.CENTER);
+        headerStyle.setVerticalAlignment(VerticalAlignment.CENTER);
+        headerStyle.setBorderBottom(BorderStyle.THIN);
+        headerStyle.setBottomBorderColor(HSSFColor.HSSFColorPredefined.TEAL.getIndex());
+        headerStyle.setWrapText(true);
+
+        Row header = sheet.createRow(0);
+        Cell headerCell = header.createCell(0);
+        headerCell.setCellValue("Report ID");
+        headerCell.setCellStyle(headerStyle);
+
+        headerCell = header.createCell(1);
+        headerCell.setCellValue("Start date");
+        headerCell.setCellStyle(headerStyle);
+
+        headerCell = header.createCell(2);
+        headerCell.setCellValue("End date");
+        headerCell.setCellStyle(headerStyle);
+    }
+
+    private void cellNumberStyle(CellStyle cellStyle){
+        cellStyle.setWrapText(true);
+        cellStyle.setBorderBottom(BorderStyle.THIN);
+        cellStyle.setDataFormat((short) 1);
+        cellStyle.setAlignment(HorizontalAlignment.LEFT);
+        cellStyle.setBottomBorderColor(HSSFColor.HSSFColorPredefined.TEAL.getIndex());
+    }
+
+    private void cellDateStyle(CellStyle cellStyle){
+        cellStyle.setWrapText(true);
+        cellStyle.setBorderBottom(BorderStyle.THIN);
+        cellStyle.setDataFormat((short) 14);
+        cellStyle.setBottomBorderColor(HSSFColor.HSSFColorPredefined.TEAL.getIndex());
+    }
+
+    public String extractAllReports() throws IOException {
+        List<Report> reportList = reportRepository.findAll();
+
+        Workbook workbook = new XSSFWorkbook();
+        Sheet sheet = workbook.createSheet("List of reports");
+        sheetStyle(sheet);
+
+        CellStyle headerStyle = workbook.createCellStyle();
+        headerCellStyle(workbook, sheet, headerStyle);
+
+        CellStyle numberStyle = workbook.createCellStyle();
+        cellNumberStyle(numberStyle);
+
+        CellStyle dateStyle = workbook.createCellStyle();
+        cellDateStyle(dateStyle);
+
+        reportList.stream().forEach(report -> writeDataInExcel(sheet, reportList.indexOf(report), report, numberStyle, dateStyle));
```

Comment:
Let's replace this forEach with indexed for-loop to have access to index and not use `indexOf`